src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -2580,6 +2580,36 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::FromOtDnsRespons
     return CHIP_NO_ERROR;
 }
 
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::ResolveAddress(intptr_t context, otDnsAddressCallback callback)
+{
+    DnsResult * dnsResult = reinterpret_cast<DnsResult *>(context);
+
+    ThreadStackMgrImpl().LockThreadStack();
+
+    const otDnsQueryConfig * defaultConfig = otDnsClientGetDefaultConfig(ThreadStackMgrImpl().OTInstance());
+
+    char fullHostName[chip::Dnssd::kHostNameMaxLength + 1 + SrpClient::kDefaultDomainNameSize + 1];
+    snprintf(fullHostName, sizeof(fullHostName), "%s.%s", dnsResult->mMdnsService.mHostName, SrpClient::kDefaultDomainName);
+
+    otDnsClientResolveAddress(ThreadStackMgrImpl().OTInstance(), fullHostName, callback, reinterpret_cast<void *>(dnsResult),
+                              defaultConfig);
```

Comment:
This is ok as is, but we can simplify and pass `NULL` as config in `otDnsClientResolveAddress()` (which will then use the default config from `otDnsClientGetDefaultConfig()`).